internal/testhelpers/e2e/session.go

```diff
@@ -189,6 +189,12 @@ func new(t *testing.T, retainDirs, updatePath bool, extraEnv ...string) *Session
 		require.NoError(session.T, err)
 	}
 
+	if runtime.GOOS == "windows" {
+		if err := cfg.Set(subshell.ConfigKeyShell, "cmd.exe"); err != nil {
+			require.NoError(session.T, err)
+		}
+	}
```

Comment:
Isn't this redundant with the env change? Also pretty sure the `cfg` instance you're operating on here isn't the one that the invoked state tool would be using, unless they happen to be the same version.